isLoading ComboBox tests and useAsyncList loadingState update (#1481)
adobe/react-spectrum: packages/@react-stately/data/docs/useAsyncList.mdx

});
```

## Filtering

There are many instances where your list of data may need to be filtered, such as during user lookup or query searches.

> For server side filtering, this can be implemented by leveraging the `filterText` passed to the `load` as an API parameter.

Reviewer: ```suggestion
For server side filtering, this can be implemented by using the `filterText` option passed to the `load` function.
```

`useAsyncList` then returns the state variable `filterText` and its setter `setFilterText`, allowing you to update the
`filterText` used within `load` in response to any updates within your component.

> 

Author: Wonder if I should include a client side filtering section...

Probably would be along the lines of "Client side filtering can be done by filtering the items returned by `useAsyncList` yourself." with an example

> `useAsyncList` returns the state variable `filterText` and its setter `setFilterText`, allowing you to update the
> `filterText` used within `load` in response to any updates within your component.

Reviewer: ```suggestion
The `setFilterText` method updates the current `filterText` value and triggers the `load` function. This allows
you to reload the results with the new filter text.
```

### Server side filtering

The example below shows how server side filtering could be implemented by using `filterText` in the `load` function and passing a parameter to the API.
The input text of the ComboBox is controlled by providing `list.filterText` as the ComboBox's `inputValue` prop. By passing `list.setFilterText` to

> the ComboBox via the `onInputChange` prop, the `filterText` within `load` is updated whenever the end user modifies the ComboBox's

Reviewer: I don't think we need to describe `setFilterText` again since we already explained it above.

> The input text of the ComboBox is controlled by providing `list.filterText` as the ComboBox's `inputValue` prop. By passing `list.setFilterText` to
> the ComboBox via the `onInputChange` prop, the `filterText` within `load` is updated whenever the end user modifies the ComboBox's
> text, ensuring that the list of items is properly updated. Finally, the state of the list (`loading`, `loadingMore`, `filtering`, etc) is
> communicated to the ComboBox through `list.state`, allowing the ComboBox to display the appropriate visual feedback to an end user.

Reviewer: ```suggestion
The input value of the ComboBox is controlled by providing `list.filterText` as the ComboBox's `inputValue` prop, and `list.setFilterText` as the `onInputChange` prop. The `loadingState` prop is also used to show the appropriate
loading indicator depending on the state of the list.
```

```tsx
let list = useAsyncList({
  async load({signal, cursor, filterText}) {
    if (cursor) {
      cursor = cursor.replace(/^http:\/\//i, 'https://');
    }

    // If no cursor is available, then we're loading the first page,
    // filtering the results returned via a query string that
    // mirrors the ComboBox input text.
    // Otherwise, the cursor is the next URL to load,
    // as returned from the previous page.

>     let res = await fetch(cursor || `https://swapi.dev/api/people/?search=${filterText}`, {signal});

Reviewer: I think we could remove the cursor stuff and loadMore stuff from this example to simplify it and keep it focused on filtering specifically.

    let json = await res.json();

    return {
      items: json.results,
      cursor: json.next
    };
  }
});

<ComboBox
  label="Star Wars Character Lookup"
  items={list.items}
  inputValue={list.filterText}
  onInputChange={list.setFilterText}

>   loadingState={list.state}

Reviewer: ```suggestion
  loadingState={list.loadingState}
```

Author: `useAsyncList` returns `state` hence the `list.state`

Reviewer: oh, hmm. that's confusing. should they match? 🤔

Author: sounds good to me. Much more descriptive that way IMO, I'll make the change
